### PR TITLE
[java] New rule LambdaCanBeMethodReference

### PR DESCRIPTION
**Pull request by oowekyala**
## Describe the PR

A new rule that should be useful with lambdas...

I would suggest we put this in the quickstart ruleset with the default properties, as it is not prone to FPs.

## Ready?

- [ ] Added unit tests for fixed bug/feature
- [ ] Passing all unit tests
- [ ] Complete build `./mvnw clean verify` passes (checked automatically by github actions)
- [ ] Added (in-code) documentation (if needed)

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 0 violations,<br />
introduces 221 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://github.com/pmd/pmd/actions/runs/8541862134">Download full report as build artifact</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by adangel — COMMENTED** [review]
Looks good, thanks!

Just some minor changes around the deprecated methods: We should directly refactor our code to not use the deprecated methods ourselves, so that we can simply cleanup by deleting these deprecated methods later.

There is one possible false-positive - this is when the lambda parameter has an annotation, e.g.:
https://github.com/pmd/java-regression-tests/blob/main/src/main/java/net/sourceforge/pmd/java/regression/tests/java11/LocalVariableSyntaxForLambdaParameters.java#L28

I'm actually ok flagging this, but still wanted to bring this up. The annotation can either be useful/necessary or not - but we can't know. If we would do a autofix however, then we should probably not change such lambdas.

> I would suggest we put this in the quickstart ruleset with the default properties, as it is not prone to FPs.

Yes, please add it to the quickstart ruleset. I like the rule :+1:

File: docs/pages/release_notes.md [hunk 1]
```diff
@@ -14,13 +14,19 @@ This is a {{ site.pmd.release_type }} release.
 
 ### 🚀 New and noteworthy
 
+#### New rules
```
Comment (adangel) [inline docs/pages/release_notes.md:17]:
```suggestion
### ✨ New rules
```

**Comment by jsotuyod** [comment]
> https://github.com/pmd/java-regression-tests/blob/main/src/main/java/net/sourceforge/pmd/java/regression/tests/java11/LocalVariableSyntaxForLambdaParameters.java#L28

Well, that *could* be rewritten as

```java
Function<@Nonnull Integer, String> lambda1 = String::valueOf;
```

which is not **exactly** the same, but is actually stronger in ensuring that the lambda is not used with nullables… So yeah, I'm ok with flagging that too.

**Review by adangel — APPROVED** [review]

**Pull request merged**